freespace2/freespace.cpp

```diff
@@ -2722,23 +2607,20 @@ extern void render_shields();
 void player_repair_frame(float frametime)
 {
 	if(MULTIPLAYER_MASTER){
-		int idx;
-		for(idx=0;idx<MAX_PLAYERS;idx++){
-			net_player *np;
+		for(auto & idx : Net_players){
+			net_player *np = &idx;
 
-			np = &Net_players[idx];
-
-			if(MULTI_CONNECTED(Net_players[idx]) && (Net_player != NULL) && (Net_player->player_id != Net_players[idx].player_id) && (Net_players[idx].m_player != NULL) && (Net_players[idx].m_player->objnum >= 0) && (Net_players[idx].m_player->objnum < MAX_OBJECTS)){
+			if(MULTI_CONNECTED(idx) && (Net_player != nullptr) && (Net_player->player_id != idx.player_id) && (idx.m_player != nullptr) && (idx.m_player->objnum >= 0) && (idx.m_player->objnum < MAX_OBJECTS)){
 
 				// don't rearm/repair if the player is dead or dying/departing
 				if ( !NETPLAYER_IS_DEAD(np) && !(Ships[Objects[np->m_player->objnum].instance].is_dying_or_departing()) ) {
```

Comment:
```suggestion
				if ( !NETPLAYER_IS_DEAD(&idx) && !(Ships[Objects[idx.m_player->objnum].instance].is_dying_or_departing()) ) {
```
And then either get rid of `np` or rename `idx` to `np`.

Reply:
You still have the now-unused `np` variable.

Reply:
yes, I'll fix it together with goobers remarks, when the kid is in bed.